lib/bacnet-asn1.js

```diff
@@ -699,11 +700,15 @@ var bacapp_decode_application_data = module.exports.bacapp_decode_application_da
       var len = result.len;
       result = bacapp_decode_data(buffer, offset + len, maxOffset, result.tagNumber, result.value);
       if (!result) return;
-      return {
+      var resObj = {
         len: len + result.len,
         type: result.type,
         value: result.value
       };
+      if (result.encoding !== undefined) {
```

Comment:
True, **if** we want to handle '0', this won't work. However, we also have to take care of 'null' then as it's equally often used.

Also as written below, I don't think it's a nice thing to have redundant and string specific checks all over the place, especially since octet-string and bit-string are working exactly the other way and utilizing an object as value.